apps/builder/app/builder/features/settings-panel/props-section/animation/animation-keyframes.tsx

```diff
@@ -157,25 +170,49 @@ const Keyframe = ({
         </Tooltip>
       </Grid>
       <Grid>
-        <CodeEditor
-          lang="css-properties"
-          size="keyframe"
-          value={cssProperties}
-          onChange={() => {
-            /* do nothing */
+        <CssEditor
+          showSearch={false}
+          propertiesPosition="top"
+          virtualize={false}
+          styleMap={styleMap}
+          apiRef={apiRef}
+          onAddDeclarations={(addedStyleMap) => {
+            const styles = { ...value.styles };
+            for (const [property, value] of addedStyleMap) {
+              styles[property] = value;
+            }
+            onChange({ ...value, styles });
+          }}
+          onDeleteProperty={(property, options = {}) => {
+            if (options.isEphemeral === true) {
+              return;
+            }
+            const styles = { ...value.styles };
+            delete styles[property];
+            onChange({ ...value, styles });
           }}
-          onChangeComplete={(cssText) => {
-            const parsedStyles = parseCss(`selector{${cssText}}`);
-            onChange({
-              ...value,
-              styles: parsedStyles.reduce(
-                (r, { property, value }) => ({ ...r, [property]: value }),
-                {}
-              ),
-            });
+          onSetProperty={(property) => {
+            return (newValue) => {
+              const styles = { ...value.styles, [property]: newValue };
+              onChange({ ...value, styles });
+            };
+          }}
+          onDeleteAllDeclarations={() => {
+            onChange({ ...value, styles: {} });
           }}
         />
       </Grid>
+      <Box css={{ paddingInline: theme.panel.paddingInline }}>
+        <Button
+          onClick={() => {
+            apiRef.current?.showAddStyleInput();
```

Comment:
😢

Reply:
this imperative style in this case is way better than all the logic mess with managing external and internal state, because that's a true mess when you need to trigger showing the element internally and externally

Reply:
I mean, invokers would here well 😸